exercises.py

```diff
@@ -3,26 +3,55 @@
 """
 
 def reverse(s: str) -> str:
-    """
-    Reverse string using recursive approach
-    """
+    if len(s) <= 1:
+        return s
+    return reverse(s[1:]) + s[0]
 
 def fib_rec(n: int) -> int:
-    """
-    Recursive fibonacci
-    """
+    if n == 0 or n == 1:
+        return 1
+    return fib_rec(n - 1) + fib_rec(n-2)
+
+
+def memoize(f):
+    cache = {}
+    def helper(num):
+        if num not in cache:            
+            cache[num] = f(num)
+        return cache[num]
+    return helper
```

Comment:
:+1: Great way to keep your global scope clean!! Since one would called `memoize`, don't you have to invoke your helper with your parameter?